mds/models.py

```diff
@@ -109,21 +109,16 @@ def __str__(self):
 
 
 class DeviceQueryset(models.QuerySet):
-    def with_latest_event(self):
-        return self.prefetch_related(
-            Prefetch(
-                "event_records",
-                queryset=EventRecord.objects.filter(
-                    id__in=Subquery(
-                        EventRecord.objects.filter(device_id=OuterRef("device_id"))
-                        .exclude(event_type="telemetry")
-                        .order_by("-timestamp")
-                        .values_list("id", flat=True)[:1]
-                    )
-                ),
-                to_attr="_latest_event",
-            )
+    def with_latest_events(self):
+        prefetchedEvents = Prefetch(
+            "event_records",
+            queryset=EventRecord.objects.exclude(event_type="telemetry").order_by(
```

Comment:
That may be the opportunity to comment why we exclude this one.

Reply:
+1 : it's because MDS Agency separates event from telemetry and the "latest_event" does not count telemetries as events.